frame/session/benchmarking/src/lib.rs

```diff
@@ -42,16 +42,33 @@ benchmarks! {
 
 	set_keys {
 		let n in 1 .. MAX_NOMINATIONS as u32;
-		let validator = create_validator_with_nominators::<T>(n, MAX_NOMINATIONS as u32)?;
+		let v_stash = create_validator_with_nominators::<T>(n, MAX_NOMINATIONS as u32)?;
+		let v_controller = pallet_staking::Module::<T>::bonded(&v_stash).ok_or("not stash")?;
 		let keys = T::Keys::default();
 		let proof: Vec<u8> = vec![0,1,2,3];
-	}: _(RawOrigin::Signed(validator), keys, proof)
+	}: _(RawOrigin::Signed(v_controller), keys, proof)
 
 	purge_keys {
 		let n in 1 .. MAX_NOMINATIONS as u32;
-		let validator = create_validator_with_nominators::<T>(n, MAX_NOMINATIONS as u32)?;
+		let v_stash = create_validator_with_nominators::<T>(n, MAX_NOMINATIONS as u32)?;
+		let v_controller = pallet_staking::Module::<T>::bonded(&v_stash).ok_or("not stash")?;
 		let keys = T::Keys::default();
 		let proof: Vec<u8> = vec![0,1,2,3];
-		Session::<T>::set_keys(RawOrigin::Signed(validator.clone()).into(), keys, proof)?;
-	}: _(RawOrigin::Signed(validator))
+		Session::<T>::set_keys(RawOrigin::Signed(v_controller.clone()).into(), keys, proof)?;
+	}: _(RawOrigin::Signed(v_controller))
 }
+
+// #[cfg(test)]
+// mod tests {
+// 	use super::*;
+// 	use pallet_session::mock::{new_test_ext, Test};
+// 	use frame_support::assert_ok;
+
+// 	#[test]
+// 	fn test_benchmarks() {
+// 		new_test_ext().execute_with(|| {
+// 			assert_ok!(test_benchmark_set_keys::<Test>());
+// 			assert_ok!(test_benchmark_purge_keys::<Test>());
+// 		});
+// 	}
+// }
```

Comment:
For some reason, this code is unable to find `palle_session::mock`... why?

Reply:
You cannot access resources behind `#[cfg(test)]`, so I had to create a new mock for just the benchmarking setup.